### DISCUSSION TITLE
/articles/sharepoint-framework-fix-spfx-error-cannot-find-module/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#243] [Comments]
Learn to fix the SharePoint Framework error ‘Cannot find module node:os’. Understand the root cause and the simple solution: running npm install.
https://www.andrewconnell.com/articles/sharepoint-framework-fix-spfx-error-cannot-find-module/

**Comment by omkarpg-maai** [comment]
I am using package tedious for sequelize to connect with ms sql server.

I have installed node modules.
Still getting same error.

Tried uninstalling all node versions and install again(using nvm)
still getting same error

Tried upgrading the packages to latest versions. 
still getting the same error.

Tried with different node versions by clearing npm cache, installing node modules again.
still getting same error

**Reply by andrewconnell** [reply]
Can you provide some context? Things like the specific error message you're receiving, the **package.json** contents, etc. all help provide context into what you're seeing to better respond and help...

**Comment by ishaisagi-hns** [comment]
Hi Andrew-  long time!
I am getting the same error regardless of the solutions offered here.
I am trying to build the v3 of the pnp modern search (as v4 will not work on premise).
I have switched between node 8, 10 & 12, each time deleting the node modules and then reinstalling - and each time getting the exact same error when doing a gulp command (either clean or build).

After troubleshooting I checked and the version of gulp I have installed is CLI 3 and local version 4.

long story short - I had to uninstall gulp, switch to Node12 (there is no way of knowing which one is the correct one without trial and error), reinstall CLI version: 3.0.0 , Local version: 4.0.2
delete the node modules and run npm install
npm install @azure/logger@1.0.0 --save-dev
npm install graceful-fs@4.2.10
npm rebuild node-sass

finally works!
